File: packages/remote-config/ios/Plugin/Classes/Events/AddConfigUpdateListenerOptionsCallbackEvent.swift
import Foundation
import Capacitor
import FirebaseRemoteConfig

@objc public class AddConfigUpdateListenerOptionsCallbackEvent: NSObject, Result {
    private var configUpdate: RemoteConfigUpdate

    init(configUpdate: RemoteConfigUpdate) {
        self.configUpdate = configUpdate
    }

    public func toJSObject() -> AnyObject {
        var updatedKeysResult = JSArray()
        for key in configUpdate.updatedKeys {
            updatedKeysResult.append(key)
        }

        var result = JSObject()
        result["updatedKeys"] = updatedKeysResult
        return result as AnyObject
    }
}

File: packages/remote-config/android/src/main/java/app/capgo/capacitor/firebase/remoteconfig/AddConfigUpdateListenerOptionsCallbackEvent.java
package app.capgo.capacitor.firebase.remoteconfig.classes.events;

import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.google.firebase.remoteconfig.ConfigUpdate;
import app.capgo.capacitor.firebase.remoteconfig.interfaces.Result;
import java.util.Set;

public class AddConfigUpdateListenerOptionsCallbackEvent implements Result {

    private ConfigUpdate configUpdate;

    public AddConfigUpdateListenerOptionsCallbackEvent(ConfigUpdate configUpdate) {
        this.configUpdate = configUpdate;
    }

    public JSObject toJSObject() {
        JSArray updatedKeysResult = new JSArray();
        for (String key : configUpdate.getUpdatedKeys()) {
            updatedKeysResult.put(key);
        }

        JSObject result = new JSObject();
        result.put("updatedKeys", updatedKeysResult);
        return result;
    }
}
